source/cgal/intersections.cc

```diff
@@ -0,0 +1,737 @@
+#  include <CGAL/Exact_predicates_exact_constructions_kernel_with_sqrt.h>
+#  include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#  include <CGAL/Kernel_traits.h>
+#  include <CGAL/Polygon_2.h>
+#  include <CGAL/Polygon_with_holes_2.h>
+#  include <CGAL/Projection_traits_xy_3.h>
+#  include <CGAL/Segment_3.h>
+#  include <CGAL/Simple_cartesian.h>
+#  include <CGAL/Tetrahedron_3.h>
+#  include <CGAL/Triangle_2.h>
+#  include <CGAL/Triangle_3.h>
+#  include <CGAL/Triangulation_2.h>
+#  include <CGAL/Triangulation_3.h>
+#  include <CGAL/Triangulation_face_base_with_id_2.h>
+#  include <CGAL/Triangulation_face_base_with_info_2.h>
+#  include <deal.II/cgal/utilities.h>
+
+#  include <fstream>
+#  include <type_traits>
+
+using K           = CGAL::Exact_predicates_exact_constructions_kernel_with_sqrt;
+using K_exact     = CGAL::Exact_predicates_exact_constructions_kernel;
+using K_inexact   = CGAL::Exact_predicates_inexact_constructions_kernel;
+using CGALPolygon = CGAL::Polygon_2<K>;
+using Polygon_with_holes_2   = CGAL::Polygon_with_holes_2<K>;
+using CGALTriangle2          = K::Triangle_2;
+using CGALTriangle3          = K::Triangle_3;
+using CGALTriangle3_exact    = K_exact::Triangle_3;
+using CGALPoint2             = K::Point_2;
+using CGALPoint3             = K::Point_3;
+using CGALPoint3_exact       = K_exact::Point_3;
+using CGALSegment2           = K::Segment_2;
+using Surface_mesh           = CGAL::Surface_mesh<K_inexact::Point_3>;
+using CGALSegment3           = K::Segment_3;
+using CGALSegment3_exact     = K_exact::Segment_3;
+using CGALTetra              = K::Tetrahedron_3;
+using CGALTetra_exact        = K_exact::Tetrahedron_3;
+using Triangulation2         = CGAL::Triangulation_2<K>;
+using Triangulation3         = CGAL::Triangulation_3<K>;
+using Triangulation3_exact   = CGAL::Triangulation_3<K_exact>;
+using Triangulation3_inexact = CGAL::Triangulation_3<K_inexact>;
+
+struct FaceInfo2
+{
+  FaceInfo2()
+  {}
+  int nesting_level;
+  bool
+  in_domain()
+  {
+    return nesting_level % 2 == 1;
+  }
+};
+
+using Vb            = CGAL::Triangulation_vertex_base_2<K>;
+using Fbb           = CGAL::Triangulation_face_base_with_info_2<FaceInfo2, K>;
+using CFb           = CGAL::Constrained_triangulation_face_base_2<K, Fbb>;
+using Fb            = CGAL::Delaunay_mesh_face_base_2<K, CFb>;
+using Tds           = CGAL::Triangulation_data_structure_2<Vb, Fb>;
+using Itag          = CGAL::Exact_predicates_tag;
+using CDT           = CGAL::Constrained_Delaunay_triangulation_2<K, Tds, Itag>;
+using Criteria      = CGAL::Delaunay_mesh_size_criteria_2<CDT>;
+using Vertex_handle = CDT::Vertex_handle;
+using Face_handle   = CDT::Face_handle;
+
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace CGALWrappers
+{
+  namespace internal
+  {
+    void
+    mark_domains(CDT &                 ct,
+                 Face_handle           start,
+                 int                   index,
+                 std::list<CDT::Edge> &border)
```

Comment:
Where is `CDT` coming from? Is it part of cgal?

Reply:
Yes it's part of cgal, it stands for Constrained Delaunay Triangulation